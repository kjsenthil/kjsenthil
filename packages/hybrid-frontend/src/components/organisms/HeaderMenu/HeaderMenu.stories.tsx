import * as React from 'react';
import { Meta, Story } from '@storybook/react/types-6-0';
import HeaderMenu, { HeaderMenuProps } from './HeaderMenu';

export default {
  title: 'Digital Hybrid/Organisms/Header Menu',
  component: HeaderMenu,
} as Meta;

const Template: Story<HeaderMenuProps> = (args) => <HeaderMenu {...args} />;

export const Default = Template.bind({});

const defaultArgs: HeaderMenuProps = {
  profileName: 'Ava Garcia',
};

Default.args = defaultArgs;
