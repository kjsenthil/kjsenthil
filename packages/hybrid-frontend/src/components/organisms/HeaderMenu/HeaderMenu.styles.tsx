import styled from 'styled-components';
import { AppBar, Link } from '../../atoms';

export const StyledAppBar = styled(AppBar)`
  box-shadow: 1px 2px 44px 0 rgba(139, 139, 139, 0.26);
`;

export const LogoImage = styled(Link)`
  ${({ theme }) => `
    padding-top: ${theme.spacing(3)}px;
    padding-right: ${theme.spacing(7)}px;
    padding-bottom: ${theme.spacing(1)}px;
    padding-left: ${theme.spacing(8)}px;

    background-image: url('/BlueBestInvest.svg');
    background-repeat: no-repeat;
    background-size: auto;
  `}
`;
