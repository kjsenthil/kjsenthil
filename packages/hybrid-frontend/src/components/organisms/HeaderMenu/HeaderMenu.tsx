import { useMediaQuery } from '@material-ui/core';
import { navigate } from 'gatsby';
import React, { useState } from 'react';
import { useDispatch } from 'react-redux';

import { logout } from '../../../services/auth';
import {
  Toolbar,
  Button,
  Grid,
  Menu,
  MenuItem,
  Icon,
  Box,
  Typography,
  Spacer,
  MenuIcon,
  IconButton,
  Link,
} from '../../atoms';
import NavLink from '../../molecules/NavLink';
import { LogoImage, StyledAppBar } from './HeaderMenu.styles';

export interface HeaderMenuProps {
  profileName: string;
}

const HeaderMenu = ({ profileName }: HeaderMenuProps) => {
  const dispatch = useDispatch();

  const isLargerScreen = useMediaQuery('(min-width: 800px)');

  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
  const [profileAnchorEl, setProfileAnchorEl] = useState<null | HTMLElement>(null);

  const handleProfileMenuClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setProfileAnchorEl(event.currentTarget);
  };

  const handleProfileMenuClose = () => {
    setProfileAnchorEl(null);
  };

  const handleMenuClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleMenuClose = () => {
    setAnchorEl(null);
  };

  const logoutHandler = () => {
    dispatch(logout());
  };

  const navigateLifePlan = () => navigate('/my-account/life-plan');
  const navigateAccounts = () => navigate('/my-account/accounts');

  const navigateHome = () => {
    navigate('/my-account/');
  };

  return (
    <Box data-test-id="header-menu">
      <StyledAppBar position="relative" data-testid="header-menu" color="inherit" elevation={0}>
        <Toolbar variant="dense">
          <Grid container justify="space-between" alignItems="center">
            <Grid item xs={12} md={8}>
              <Grid container justify="space-evenly" alignItems="center">
                <Grid item xs={4}>
                  <Link onClick={navigateHome}>
                    <LogoImage />
                  </Link>
                </Grid>

                <Spacer x={3} />

                {!isLargerScreen && (
                  <Grid item xs={5}>
                    <Grid container justify="flex-end">
                      <IconButton
                        edge="start"
                        color="primary"
                        aria-label="menu"
                        onClick={handleMenuClick}
                      >
                        <MenuIcon />
                      </IconButton>
                      <Menu
                        id="simple-menu"
                        anchorEl={anchorEl}
                        keepMounted
                        open={Boolean(anchorEl)}
                        onClose={handleMenuClose}
                      >
                        <MenuItem onClick={navigateHome}>Home</MenuItem>
                        <MenuItem onClick={navigateAccounts}>Investments</MenuItem>
                        <MenuItem onClick={navigateLifePlan}>Life Plan</MenuItem>
                        <MenuItem onClick={logoutHandler}>Logout</MenuItem>
                      </Menu>
                    </Grid>
                  </Grid>
                )}

                {isLargerScreen && (
                  <>
                    <Grid item>
                      <NavLink onClick={navigateHome}>
                        <Typography variant="sh4" color="grey">
                          Home
                        </Typography>
                      </NavLink>
                    </Grid>
                    <Grid item>
                      <NavLink onClick={() => navigate('/my-account/accounts')}>
                        <Typography variant="sh4" color="grey">
                          Investments
                        </Typography>
                      </NavLink>
                    </Grid>
                    <Grid item>
                      <NavLink onClick={navigateLifePlan}>
                        <Typography variant="sh4" color="grey">
                          Life Plan
                        </Typography>
                      </NavLink>
                    </Grid>
                  </>
                )}
              </Grid>
            </Grid>

            {isLargerScreen && (
              <Grid item xs={12} md={4}>
                <Grid container justify="flex-end">
                  <Grid item>
                    <Grid container>
                      <Button
                        aria-controls="simple-menu"
                        aria-haspopup="true"
                        variant={undefined}
                        color="primary"
                        onClick={handleProfileMenuClick}
                        endIcon={<Icon name="arrowHeadDown" />}
                      >
                        {profileName}
                      </Button>
                    </Grid>
                    <Menu
                      id="simple-menu"
                      anchorEl={profileAnchorEl}
                      keepMounted
                      open={Boolean(profileAnchorEl)}
                      onClose={handleProfileMenuClose}
                    >
                      <MenuItem onClick={logoutHandler}>Logout</MenuItem>
                    </Menu>
                  </Grid>
                </Grid>
              </Grid>
            )}
          </Grid>
        </Toolbar>
      </StyledAppBar>
    </Box>
  );
};

export default HeaderMenu;
