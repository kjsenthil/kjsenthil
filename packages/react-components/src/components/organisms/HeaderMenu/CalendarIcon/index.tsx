import styled from 'styled-components';
import * as React from 'react';
import { useTheme } from '@material-ui/core';
import { Icon } from '../../../atoms';

const CalendarIconContainer = styled.div`
  position: relative;
  left: 30px;
  top: 27px;
`;

type PaletteType = {
  grey: {
    light2: string;
  };
};

const CalendarIcon = () => {
  const { palette } = useTheme();
  return (
    <CalendarIconContainer>
      <svg width="20" height="20" viewBox="0 0 20 20">
        <g fill="none" fillRule="evenodd">
          <g transform="translate(-1384.000000, -133.000000) translate(0.000000, 90.000000) translate(1216.000000, 17.000000) translate(168.000000, 26.000000)">
            <circle
              cx="10"
              cy="10"
              r="10"
              fill={((palette as unknown) as PaletteType).grey.light2}
            />
            <g transform="scale(0.8, 0.8) translate(2.5 2.5)">
              <Icon name="calendar" color="primary" />
            </g>
          </g>
        </g>
      </svg>
    </CalendarIconContainer>
  );
};

export default CalendarIcon;
