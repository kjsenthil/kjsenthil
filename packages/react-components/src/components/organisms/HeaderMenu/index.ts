import HeaderMenu, { HeaderMenuProps } from './HeaderMenu';

export type { HeaderMenuProps };
export default HeaderMenu;
