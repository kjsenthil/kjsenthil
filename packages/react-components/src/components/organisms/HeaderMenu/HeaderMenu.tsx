import React, { useState } from 'react';
import Img from 'gatsby-image';
import {
  Box,
  Button,
  ClearIcon,
  Divider,
  Drawer,
  Grid,
  Icon,
  InputAdornment,
  Link,
  ListItem,
  ListItemSecondaryAction,
  MenuIcon,
  Switcher,
  useMediaQuery,
  useTheme,
  Typography,
} from '../../atoms';
import { DisabledComponent, NavLink, SubHeader } from '../../molecules';
import { ClickAwayListener } from '../../particles';
import CoachingPopover from '../CoachingPopover';
import {
  Availability,
  CallBack,
  CallContainer,
  CashAndInvestGrid,
  Circle,
  CoachIcon,
  CoachIconContainer,
  CoachTextContainer,
  DrawerContainer,
  DrawerFooterContainer,
  IconGridDivider,
  InvestButton,
  LogoImage,
  ModalContainer,
  MyAccountsContainer,
  ProfileMenuItem,
  RotatedIcon,
  StyledAppBar,
  StyledCallBackLink,
  StyledMenuNavGrid,
  StyledMobileMenuList,
  StyledIconButton,
  StyledInputBase,
  StyledProfileList,
  StyledSubHeader,
  StyledToolbar,
  SwitcherLabel,
} from './HeaderMenu.styles';
import { StyledChildNavLink } from '../../molecules/NavLinkWithDropDown/NavLinkWithDropDown.styles';
import Spacer from '../../atoms/Spacer/Spacer';
import NavLinkWithDropDown, { NavLinkWithDropDownProps } from '../../molecules/NavLinkWithDropDown';
import CalendarIcon from './CalendarIcon';

type LinkWithSwitch = {
  onClick: (isEnabled: boolean) => void;
  type: 'switch';
};

type LinkWithPath = {
  path: string;
  type?: string;
};

type LinkType = LinkWithSwitch | LinkWithPath;

export type LinkData = {
  name: string;
  shouldShowInMainMenu: boolean;
  shouldShowInDrawer: boolean;
  color?: 'primary' | 'error';
  icon?: React.ReactElement;
  disabled?: boolean;
  type?: string;
  childLinks?: {
    path: string;
    name: string;
    disabled?: boolean;
  }[];
} & LinkType;

type GatsbyImage = {
  childImageSharp: {
    fluid: {
      aspectRatio: number;
      src: string;
      srcSet: string;
      sizes: string;
    };
  };
};

type CoachImages = {
  coachPortrait: GatsbyImage;
  coachIcon: GatsbyImage;
};

export interface HeaderMenuProps {
  homePath: string;
  links: Array<LinkData>;
  switchHandler: (evt: React.ChangeEvent<HTMLInputElement>) => void;
  isExpFeatureFlagEnabled?: boolean;
  currentUrl: string;
  isNonProd?: boolean;
  myAccountsUrl?: string;
  coachImages: CoachImages;
  showCoachPopover: boolean;
  toggleCalendlyModal: () => void;
  toggleCoachPopover: (value?: boolean) => void;
  navigate: (path: string) => {};
}

const HeaderMenu = ({
  currentUrl,
  homePath,
  links,
  isNonProd = true,
  switchHandler,
  isExpFeatureFlagEnabled = true,
  myAccountsUrl,
  coachImages,
  toggleCalendlyModal,
  showCoachPopover,
  toggleCoachPopover,
  navigate,
}: HeaderMenuProps) => {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
  const [profileMenuOpen, setProfileMenuOpen] = useState<boolean>(false);
  const [menuOpen, setMenuOpen] = useState<boolean>(false);

  const handleProfileMenuClick = () => {
    setProfileMenuOpen(!profileMenuOpen);
  };

  const handleMenuClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
    setMenuOpen(true);
  };

  const handleMenuClose = () => {
    setAnchorEl(null);
    setMenuOpen(false);
  };

  const handlePhonelineStatus = () => {
    const today = new Date(Date.now());
    const day = today.getDay();
    const time = today.getHours();
    if (day === 0 || day === 6 || time >= 18 || time < 9) {
      return (
        <>
          <Circle data-testid="circle-phoneline-icon" color={theme.palette.error.main} />
          <Availability>We are currently unavailable</Availability>
        </>
      );
    }
    return (
      <>
        <Circle data-testid="circle-phoneline-icon" color={theme.palette.success.main} />
        <Availability>We are open, call us: 020 7189 9999</Availability>
      </>
    );
  };

  const renderSwitcher = (withInnerLabel: boolean, onClick: (evt) => void) => (
    <Switcher
      size="small"
      withInnerLabel={withInnerLabel}
      checked={isExpFeatureFlagEnabled}
      onClick={onClick}
    />
  );

  const navigateHome = () => navigate(homePath);

  const renderProfileMenuDropdownLinks = (profileLink: LinkData | undefined) =>
    profileLink?.childLinks?.map((childLink) => (
      <ProfileMenuItem
        tabIndex={0}
        key={`${childLink.name}-Profile-MenuItem`}
        onClick={() => navigate((childLink as LinkWithPath).path as string)}
      >
        <StyledChildNavLink isMobile={isMobile}>{childLink.name}</StyledChildNavLink>
      </ProfileMenuItem>
    ));

  const getOnLinkClick = (link: LinkData) =>
    link.type === 'switch'
      ? (link as LinkWithSwitch).onClick
      : () => navigate((link as LinkWithPath).path as string);

  const styledDivider = (
    <IconGridDivider>
      <Divider orientation="vertical" y={4} />
    </IconGridDivider>
  );

  const renderMyAccountsLink = () => (
    <MyAccountsContainer>
      <RotatedIcon name="upload" />
      <Link color="white" href={myAccountsUrl}>
        Back to old Bestinvest
      </Link>
    </MyAccountsContainer>
  );

  const renderCoachSignpost = () => (
    <CoachIconContainer>
      <ModalContainer>
        {showCoachPopover && (
          <ClickAwayListener mouseEvent="onMouseDown" onClickAway={() => toggleCoachPopover(false)}>
            <CoachingPopover
              image={
                <Img
                  fluid={coachImages.coachPortrait.childImageSharp.fluid}
                  alt="Portrait image of coach"
                />
              }
              onButtonClick={toggleCalendlyModal}
            />
          </ClickAwayListener>
        )}
      </ModalContainer>

      <CoachTextContainer>
        <Typography variant="sh1" color="inherit">
          YOUR COACH
        </Typography>
        <Link
          color="white"
          special
          onClick={() => (isMobile ? toggleCalendlyModal() : toggleCoachPopover())}
        >
          Book an appointment
        </Link>
      </CoachTextContainer>

      <CoachIcon coachIconUrl={coachImages.coachIcon.childImageSharp.fluid.src}>
        <CalendarIcon />
      </CoachIcon>
    </CoachIconContainer>
  );

  const disableableNavLinks = (children: React.ReactNode, index: number, disabled?: boolean) => {
    if (!disabled) {
      return <div key={`enabled-${index}`}>{children}</div>;
    }
    return (
      <DisabledComponent title="Coming soon" key={`disabled-${index}`}>
        {children}
      </DisabledComponent>
    );
  };

  const navLinkWithDropDown = (link: LinkData, mobile: boolean) => (
    <NavLinkWithDropDown
      key={`${link.name}-${mobile ? 'mobile' : 'desktop'}-menu-dropdown`}
      name={link.name}
      type={link.type as string}
      path={(link as LinkWithPath).path}
      childLinks={link.childLinks as NavLinkWithDropDownProps['childLinks']}
      selected={(link as LinkWithPath).path === currentUrl}
      navigate={navigate}
    />
  );

  const menuLink = (link: LinkData) => (
    <NavLink
      onClick={getOnLinkClick(link)}
      selected={(link as LinkWithPath).path === currentUrl}
      isMobile={isMobile}
      key={`${link.name}-desktop-menu`}
      variant="link"
    >
      {link.name}
    </NavLink>
  );

  const drawerLink = (link: LinkData) => (
    <>
      <ListItem
        key={`${link.name}-mobile-menu`}
        onClick={getOnLinkClick(link) as any}
        button={link.type === 'switch' ? false : undefined}
      >
        {!!link.icon && (
          <>
            {link.icon}
            <Spacer x={1} />
          </>
        )}

        <NavLink
          colorShade={link.color === 'error' ? undefined : 'dark2'}
          color={link.color ?? 'primary'}
          selected={(link as LinkWithPath).path === currentUrl}
          isMobile={isMobile}
          key={`${link.name}-mobile-link`}
          variant="link"
        >
          {link.name}
        </NavLink>

        {link.type === 'switch' && (
          <ListItemSecondaryAction key={`${link.name}-mobile-switch`}>
            {renderSwitcher(false, (evt) => getOnLinkClick(link)(evt.target.checked))}
          </ListItemSecondaryAction>
        )}
      </ListItem>
    </>
  );

  const renderDrawerList = () =>
    links
      .filter((link) => link.shouldShowInDrawer)
      .map((link, index) =>
        link.childLinks
          ? disableableNavLinks(navLinkWithDropDown(link, true), index, link.disabled)
          : disableableNavLinks(drawerLink(link), index, link.disabled)
      );

  const renderMenuNavLinks = () =>
    links
      .filter((link) => link.shouldShowInMainMenu)
      .map((link, index) => (
        <Grid
          item
          key={`${(link as LinkWithPath).path || link.name}-menu`}
          {...(link.disabled ? { disabled: true } : {})}
        >
          {link.childLinks
            ? disableableNavLinks(navLinkWithDropDown(link, false), index, link.disabled)
            : disableableNavLinks(menuLink(link), index, link.disabled)}
        </Grid>
      ));

  const renderCashAndInvestActions = (forSubMenu: boolean = true) => {
    const ButtonWrapper = ({ children }: { children: React.ReactElement }) =>
      forSubMenu ? (
        <Grid item container justifyContent="space-between" spacing={2} wrap="nowrap">
          {children}
        </Grid>
      ) : (
        children
      );
    return (
      <>
        <ButtonWrapper>
          <>
            <Grid item>
              <DisabledComponent title="Coming soon">
                <Button
                  data-testid="add cash"
                  color="primary"
                  wrap="nowrap"
                  startIcon={isMobile ? null : <Icon name="plus" />}
                  variant="outlined"
                >
                  Add cash
                </Button>
              </DisabledComponent>
            </Grid>

            <CashAndInvestGrid isMobile={isMobile} {...(isMobile ? { item: true, xs: 9 } : {})}>
              <DisabledComponent title="Coming soon">
                <InvestButton
                  data-testid="invest"
                  color="gradient"
                  variant="contained"
                  {...(isMobile
                    ? {}
                    : {
                        startIcon: <Icon name="investStatistics" />,
                      })}
                  isMobile={isMobile}
                >
                  Invest
                </InvestButton>
              </DisabledComponent>
            </CashAndInvestGrid>
          </>
        </ButtonWrapper>
      </>
    );
  };

  return (
    <Box data-testid="header">
      <StyledAppBar
        data-testid="header-menu"
        color="inherit"
        elevation={0}
        isMobile={isMobile}
        position="relative"
        style={{ zIndex: 1201 }}
      >
        <StyledToolbar variant="dense" isMobile={isMobile}>
          <Grid container justifyContent="space-between" alignItems="center">
            <Grid container justifyContent="space-between" direction="row" alignItems="center">
              <Grid item xs={2}>
                <Link onClick={navigateHome}>
                  <LogoImage isMobile={isMobile} />
                </Link>
              </Grid>

              {isMobile ? (
                <Grid item xs={9}>
                  <Grid container justifyContent="flex-end">
                    <StyledIconButton
                      edge="start"
                      color="primary"
                      aria-label={!menuOpen ? 'menu' : 'close menu'}
                      onClick={!menuOpen ? handleMenuClick : handleMenuClose}
                    >
                      {!menuOpen ? <MenuIcon /> : <ClearIcon />}
                    </StyledIconButton>

                    <Drawer
                      anchor="right"
                      keepMounted
                      open={Boolean(anchorEl)}
                      onClose={handleMenuClose}
                      style={{ zIndex: 1200 }} // Must be inline to override z-index set by MUI modal JS
                    >
                      <DrawerContainer>
                        {isExpFeatureFlagEnabled ? (
                          <StyledSubHeader>{renderCoachSignpost()}</StyledSubHeader>
                        ) : null}

                        <DisabledComponent title="Coming soon">
                          <StyledInputBase
                            placeholder="Markets, products, stocks"
                            startAdornment={
                              <InputAdornment position="start">
                                <Icon name="search" color="disabled" />
                              </InputAdornment>
                            }
                          />
                        </DisabledComponent>

                        <StyledMobileMenuList isExpFeatureFlagEnabled={isExpFeatureFlagEnabled}>
                          {isNonProd && (
                            <SwitcherLabel
                              labelPlacement="start"
                              control={<Box ml={1}>{renderSwitcher(true, switchHandler)}</Box>}
                              label="Experimental Features"
                            />
                          )}

                          {renderDrawerList()}
                        </StyledMobileMenuList>

                        <DrawerFooterContainer>
                          {renderCashAndInvestActions()}
                          <CallContainer>
                            {handlePhonelineStatus()}
                            <CallBack>
                              <DisabledComponent title="Coming soon">
                                <StyledCallBackLink>Request a callback</StyledCallBackLink>
                              </DisabledComponent>
                            </CallBack>
                          </CallContainer>
                        </DrawerFooterContainer>
                      </DrawerContainer>
                    </Drawer>
                  </Grid>
                </Grid>
              ) : (
                <Grid
                  item
                  container
                  direction="row"
                  justifyContent="space-between"
                  xs={10}
                  wrap="nowrap"
                >
                  <StyledMenuNavGrid item container justifyContent="flex-start" wrap="nowrap">
                    {renderMenuNavLinks()}
                  </StyledMenuNavGrid>

                  <Grid
                    item
                    container
                    spacing={2}
                    justifyContent="flex-end"
                    alignItems="center"
                    wrap="nowrap"
                    component="div"
                  >
                    {renderCashAndInvestActions(false)}

                    {styledDivider}

                    <Grid item>
                      <DisabledComponent title="Coming soon">
                        <Button
                          aria-controls="simple-menu"
                          aria-haspopup="true"
                          variant={undefined}
                          color="primary"
                          aria-label="list"
                          // onClick={handleProfileMenuClick} // To be added
                        >
                          <Icon name="list" />
                        </Button>
                      </DisabledComponent>
                    </Grid>

                    {styledDivider}

                    <Grid item>
                      <DisabledComponent title="Coming soon">
                        <Button
                          aria-controls="simple-menu"
                          aria-haspopup="true"
                          variant={undefined}
                          color="primary"
                          aria-label="search"
                          // onClick={handleProfileMenuClick} // To be added
                        >
                          <Icon name="search" />
                        </Button>
                      </DisabledComponent>
                    </Grid>

                    {styledDivider}

                    <Grid item>
                      <DisabledComponent title="Coming soon">
                        <Button
                          aria-controls="simple-menu"
                          aria-haspopup="true"
                          variant={undefined}
                          color="primary"
                          aria-label="shopping-cart"
                          // onClick={handleProfileMenuClick}  // To be added
                        >
                          <Icon name="cart" />
                        </Button>
                      </DisabledComponent>
                    </Grid>

                    {styledDivider}

                    <Grid item>
                      <Button
                        data-testid="profile"
                        aria-controls="simple-menu"
                        aria-haspopup="true"
                        variant={undefined}
                        color="primary"
                        aria-label="profile"
                        onClick={handleProfileMenuClick}
                      >
                        <Icon name="account" />
                      </Button>
                    </Grid>

                    {profileMenuOpen && (
                      <StyledProfileList profileMenuOpen={profileMenuOpen}>
                        {renderProfileMenuDropdownLinks(
                          links?.find((linkItem) => linkItem.name === 'Profile')
                        )}
                      </StyledProfileList>
                    )}
                  </Grid>
                </Grid>
              )}
            </Grid>
          </Grid>
        </StyledToolbar>

        {isExpFeatureFlagEnabled ? (
          <>
            {isMobile ? (
              !menuOpen && (
                <StyledSubHeader isMobile={isMobile}>
                  <Grid item container justifyContent="center">
                    {renderMyAccountsLink()}
                  </Grid>
                </StyledSubHeader>
              )
            ) : (
              <StyledSubHeader isMobile={isMobile}>
                <Grid container justifyContent="space-between" alignItems="center">
                  <Grid item>{renderMyAccountsLink()}</Grid>

                  <Grid item>
                    {isNonProd && (
                      <SwitcherLabel
                        labelPlacement="start"
                        control={<Box ml={1}>{renderSwitcher(true, switchHandler)}</Box>}
                        label="Experimental Features"
                      />
                    )}
                  </Grid>

                  <Grid item>{renderCoachSignpost()}</Grid>
                </Grid>
              </StyledSubHeader>
            )}
          </>
        ) : (
          !isMobile && (
            <SubHeader style={{ minHeight: '80px' }}>
              <Grid container justifyContent="space-between" alignItems="center">
                <Grid item>
                  {isNonProd && (
                    <SwitcherLabel
                      labelPlacement="start"
                      control={<Box ml={1}>{renderSwitcher(true, switchHandler)}</Box>}
                      label="Experimental Features"
                    />
                  )}
                </Grid>

                <Grid item>
                  <Button color="primary" variant="outlined" href={myAccountsUrl}>
                    My Accounts Login
                  </Button>
                </Grid>
              </Grid>
            </SubHeader>
          )
        )}
      </StyledAppBar>
    </Box>
  );
};

export default HeaderMenu;
