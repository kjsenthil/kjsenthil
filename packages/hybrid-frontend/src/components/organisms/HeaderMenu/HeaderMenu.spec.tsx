import React from 'react';
import { renderWithTheme } from '@tsw/test-util';
import HeaderMenu from './HeaderMenu';

const props = {
  cash: 'Â£101,100.00',
  homePath: '/',
  currentUrl: '/investment',
  expFeatureSwitch: () => {},
  links: [
    {
      name: 'Investment',
      path: '/investment',
    },
    { name: 'Life plan', path: '/life-plan' },
    { name: 'Logout', path: '/logout', shouldShowInDrawer: true },
  ],
};

describe('HeaderMenu', () => {
  it('renders the header menu', async () => {
    const { result } = renderWithTheme(<HeaderMenu {...props} />);
    expect(result.container).toMatchSnapshot();
  });
});
