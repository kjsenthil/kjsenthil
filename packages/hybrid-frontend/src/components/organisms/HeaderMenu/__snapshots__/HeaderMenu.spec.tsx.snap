// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu renders the header menu 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
    data-test-id="header-menu"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionRelative MuiAppBar-colorInherit sc-faUofl kpiAT MuiPaper-elevation0"
      data-testid="header-menu"
    >
      <div
        class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
            >
              <button
                class="MuiTypography-root MuiLink-root MuiLink-underlineHover sc-dJjZJu hkdYLt MuiLink-button MuiTypography-colorPrimary"
              >
                <div
                  class="sc-GamvQ eLvPWY"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-flex-end"
              >
                <button
                  aria-label="menu"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-2"
        data-testid="Spacer"
      >
        <div
          class="sc-iqsfdx YcysP"
        />
      </div>
      <div
        class="MuiToolbar-root MuiToolbar-regular sc-eJwXpk hSaTsD MuiToolbar-gutters"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-wrap-xs-nowrap MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
          >
            <div
              class="sc-dvQbkV bZbwOW"
            >
              <h6
                class="MuiTypography-root sc-jrQzUz bxceGS MuiTypography-body1"
              >
                Â£101,100.00
              </h6>
              <p
                class="MuiTypography-root sc-jrQzUz ifrOuG MuiTypography-body1"
              >
                cash ready to invest
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-wrap-xs-nowrap MuiGrid-justify-xs-flex-end"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined sc-gsDJrp jHMvjH MuiButton-disableElevation"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17.739 6.261c.396.396.471.964-.718 2.152L13.435 12l3.586 3.587c1.19 1.188 1.114 1.756.718 2.152-.396.396-.964.471-2.152-.718L12 13.435 8.413 17.02c-1.188 1.19-1.756 1.114-2.152.718-.396-.396-.471-.964.717-2.152L10.565 12 6.978 8.413C5.79 7.225 5.865 6.657 6.261 6.261c.396-.396.964-.471 2.152.717L12 10.565l3.587-3.587c1.188-1.188 1.756-1.113 2.152-.717z"
                      />
                    </svg>
                  </span>
                  Add cash
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained sc-gsDJrp fELEhA MuiButton-disableElevation"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M4 4c.552 0 1 .343 1 2v10c0 .943 0 1.414.293 1.707C5.586 18 6.057 18 7 18h.003C7 17.868 7 17.707 7 17.5v-6c0-.465 0-.697.038-.89.158-.794.778-1.414 1.572-1.572C8.803 9 9.035 9 9.5 9s.697 0 .89.038c.794.158 1.414.778 1.572 1.572.038.193.038.425.038.89v6c0 .207 0 .368-.003.5h2.006C14 17.868 14 17.707 14 17.5v-9c0-.465 0-.697.038-.89.158-.794.778-1.414 1.572-1.572C15.803 6 16.035 6 16.5 6s.697 0 .89.038c.794.158 1.414.778 1.572 1.572.038.193.038.425.038.89v9c0 .207 0 .368-.003.5H19c1.657 0 2 .448 2 1s-.343 1-2 1H9c-2.828 0-4.243 0-5.121-.879C3 18.243 3 16.828 3 14V6c0-1.657.448-2 1-2zm12.5 14c.246 0 .38 0 .479-.004h.017v-.017C17 17.88 17 17.745 17 17.5v-9c0-.246 0-.38-.004-.478v-.018h-.017C16.88 8 16.745 8 16.5 8c-.246 0-.38 0-.479.004h-.017v.018C16 8.12 16 8.254 16 8.5v9c0 .246 0 .38.004.479v.017h.017c.099.004.233.004.479.004zm-7 0c.246 0 .38 0 .478-.004h.018v-.017C10 17.88 10 17.745 10 17.5v-6c0-.246 0-.38-.004-.478v-.018h-.018C9.88 11 9.746 11 9.5 11c-.246 0-.38 0-.478.004h-.018v.018C9 11.12 9 11.254 9 11.5v6c0 .246 0 .38.004.479v.017h.018C9.12 18 9.254 18 9.5 18z"
                      />
                    </svg>
                  </span>
                  Invest
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </header>
  </div>
</div>
`;
